### PR TITLE
feat: include process id in publish-batch logs

### PR DESCRIPTION
**Pull request by bajtos**
Make it easier to understand interleaved log messages from multiple children processes running concurrently.

### CODE REVIEW

File: voyager-publish/bin/publish-batch.js [hunk 1]
```diff
@@ -68,12 +68,27 @@ const ieContract = new ethers.Contract(
   provider
 ).connect(signer)
 
+const addPidToConsoleArgs = (...args) => {
+  if (typeof args[0] === 'string') {
+    args[0] = `[worker:%s] ${args[0]}`
+    args.splice(1, 0, process.pid)
+  } else {
+    args.unshift(`[worker:${process.pid}]`)
+  }
```
Comment (juliangruber) [inline voyager-publish/bin/publish-batch.js:77]:
```suggestion
  args.unshift(`[worker:${process.pid}]`)
```

I think that should create the same result

File: voyager-publish/bin/publish-batch.js [hunk 2]
```diff
@@ -68,12 +68,27 @@ const ieContract = new ethers.Contract(
...
+  return args
+}
+const logger = {
+  log (...args) { console.log(...addPidToConsoleArgs(...args)) },
+  error (...args) { console.log(...addPidToConsoleArgs(...args)) }
```
Comment (juliangruber) [inline voyager-publish/bin/publish-batch.js:82]:
```suggestion
  error (...args) { console.error(...addPidToConsoleArgs(...args)) }
```

**Review by juliangruber — CHANGES_REQUESTED** [review]
good idea!

File: voyager-publish/bin/publish-batch.js (hunk 1 above)
Comment (juliangruber) [reply voyager-publish/bin/publish-batch.js:77]: no scratch that it doesn't, because of fprintf formatting. But I think I have a different idea

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: voyager-publish/bin/publish-batch.js (hunk 1 above)
Comment (juliangruber) [reply voyager-publish/bin/publish-batch.js:77]: see next review

File: voyager-publish/bin/publish-batch.js [hunk 3]
```diff
@@ -68,12 +68,27 @@ const ieContract = new ethers.Contract(
...
 ).connect(signer)
 
+const addPidToConsoleArgs = (...args) => {
+  if (typeof args[0] === 'string') {
+    args[0] = `[worker:%s] ${args[0]}`
+    args.splice(1, 0, process.pid)
+  } else {
+    args.unshift(`[worker:${process.pid}]`)
+  }
+  return args
```
[the remark this answers is not in the document]
Comment (bajtos) [reply voyager-publish/bin/publish-batch.js:78]: The problem with your suggestion is that `console.log` applies colours to arguments while `util.format` does not.

Screenshot 2024-05-09 at 13 13 35

**Review by juliangruber — APPROVED** [review]

**Pull request merged**